### PR TITLE
feat(web): add account page to change password

### PR DESCRIPTION
**Pull request by ZTongci**
# Overview

## What I've done
Adding page for account setting
Using hook to show userName and userEmail for account setting page
Adding interface to update account password

## What I haven't done
add function to delete account

## How I tested
Using my account, click change password. And using new password login successful.

## Release Notes

- **New Features**
  - Introduced a comprehensive account settings interface, allowing users to manage their account and workspace settings.
  - Added functionality for users to update their passwords and delete their accounts through new modal components.
  - Implemented a new hook for managing user data and password policies, enhancing account management capabilities.
  - Added a direct navigation option to account settings from the profile menu.
  - Expanded routing to include a dedicated settings page for account management.

- **Bug Fixes**
  - Enhanced password validation and user feedback during password updates.

- **Refactor**
  - Updated component structures for better organization and encapsulation within the account settings page.

- **Style**
  - Improved layout and styling for the account settings interface, ensuring a consistent user experience.

- **Localization**
  - Expanded localization support with new translation keys for account management and user notifications in both English and Japanese.

### CODE REVIEW

File: web/src/beta/features/AccountAndWorkSpaceSetting/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  const passwordPolicy = window.REEARTH_CONFIG?.passwordPolicy;
+
+  const { me: data } = useMeQuery();
+
+  const handleUpdateUserPassword = useCallback(
+    async ({ password, passwordConfirmation }: UpdatePasswordType) => {
+      await useUpdatePassword({ password, passwordConfirmation });
+    },
+    [useUpdatePassword]
+  );
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/hooks.ts]: done!
https://github.com/reearth/reearth-visualizer/pull/1155/commits/add2f4929bcd48f3b1ebeef48e120c9a1097e285

File: web/src/beta/features/AccountAndWorkSpaceSetting/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  const handleDeleteUser = useCallback(async () => {
+    const userId = data.id;
+    if (userId) await useDeleteUser({ userId });
+  }, [data.id, useDeleteUser]);
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/hooks.ts]: Already add try catch to `await useDeleteUser({ userId });`
So don't need this.

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  return (
+    <InnerPage>
+      <SettingsWrapper>
+        <Collapse size="large" title={t("Account")}>
+          <SettingsFields>
+            <InputField
+              title={t("Name")}
+              value={t(imformationData.name ? imformationData.name : "Red")}
+              appearance="readonly"
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx]: done
https://github.com/reearth/reearth-visualizer/pull/1155/commits/40e6f43ac0cff74a953860755948feab53e5e5bb

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+          </SettingsFields>
+        </Collapse>
+      </SettingsWrapper>
+
+      <PasswordModal
+        isVisible={!onPasswordModalClose}
+        passwordPolicy={passwordPolicy}
+        onClose={() => setPasswordModalOnClose(!onPasswordModalClose)}
+        updatePassword={onUpdateUserPassword}
+      />
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx:83]: use hook already added.

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import PasswordModal from "./PasswordModal";
+
+type Props = {
+  imformationData: { name?: string; email?: string };
+  passwordPolicy?: PasswordPolicy;
+  onUpdateUserPassword: ({
+    password,
+    passwordConfirmation
+  }: UpdatePasswordType) => Promise<void>;
+};
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx]: done
https://github.com/reearth/reearth-visualizer/pull/1155/commits/d18ff765626f5cdd096bc78bc643bc25512635df

File: web/src/beta/features/AccountAndWorkSpaceSetting/index.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+      accountSettingTabs.map((tab) => ({
+        id: tab.id,
+        icon: tab.icon,
+        text: t(tab.text),
+        path: `/settings/${tab.id}`
+      })),
+    [t]
+  );
+  const { meData, passwordPolicy, handleUpdateUserPassword } = useHook();
+  console.log(meData);
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/index.tsx:43]: done

File: web/src/beta/features/AccountAndWorkSpaceSetting/index.tsx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+            <SidebarVersion />
+          </SidebarWrapper>
+        </LeftSidePanel>
+        <Content>
+          {tab === "account" && (
+            <AccountSetting
+              onUpdateUserPassword={handleUpdateUserPassword}
+              passwordPolicy={passwordPolicy}
+              imformationData={{ name, email }}
+            />
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/index.tsx:76]: done

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+    setPasswordConfirmation("");
+    onClose?.();
+  }, [onClose]);
+
+  const handleSave = useCallback(() => {
+    if (password === passwordConfirmation) {
+      updatePassword?.({ password, passwordConfirmation });
+      handleClose();
+    }
+  }, [updatePassword, handleClose, password, passwordConfirmation]);
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx:102]: Already add try catch to hook

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+              </Typography>
+              <TextInput
+                value={password}
+                onChange={handlePasswordChange}
+                type="password"
+              />
+              {password ? (
+                <PasswordMessage size="body" weight="bold">
+                  {regexMessage}
+                </PasswordMessage>
```
Comment (airslice) [inline web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx:146]: I think it's better to have different colors for differenty type of regexMessage as well. (danger / warning / okey), an d please use the colors from theme.

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+              <Typography size="body" weight="bold">
+                {t("New password (for confirmation)")}
+              </Typography>
+              <TextInput
+                value={passwordConfirmation}
+                onChange={setPasswordConfirmation}
+                type="password"
+              />
+              {isMatchPassword ? (
+                <PasswordMessage size="body" weight="regular" color="red">
```
Comment (airslice) [inline web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx:159]: Please use the color from theme.

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+const ModalContentWrapper = styled("div")(({ theme }) => ({
+  display: "flex",
+  flexDirection: "column",
+  gap: theme.spacing.large,
+  padding: theme.spacing.large,
+  background: theme.bg[1]
+}));
+
+const SubText = styled.div`
+  margin: ${({ theme }) => `${theme.spacing.large}px auto`};
```
Comment (airslice) [inline web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx:183]: Please check the layout solution on design. Basically we locate this together with two password fields vertically in a flex box, and making the space between them by gap.

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/common.tsx [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+export const Thumbnail = styled("div")<{ src?: string }>(({ src, theme }) => ({
+  width: "100%",
+  paddingBottom: "52.3%",
+  fontSize: 0,
+  background: src
+    ? `url(${src}) center/contain no-repeat`
+    : theme.relative.dark,
+  borderRadius: theme.radius.small
+}));
```
Comment (airslice) [inline web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/common.tsx:60]:
```suggestion
```
Not usefull here i guess

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/common.tsx [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+  return (
+    <Collapse title={t("Notice")} size="large">
+      <Typography size="body">
+        {t(
+          "Most project settings are hidden when the project is archived. Please unarchive the project to view and edit these settings."
+        )}
+      </Typography>
+    </Collapse>
+  );
+};
```
Comment (airslice) [inline web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/common.tsx:79]:
```suggestion
```
Not usefull here i guess.

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx (hunk 11 above)
Comment (airslice) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx:183]: BTW please check all text font weight as well, i think they're not in `bold` on design.

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+
+const AccountSetting: FC<Props> = ({
+  passwordPolicy,
+  onUpdateUserPassword,
+  informationData
+}) => {
+  const t = useT();
+  const [onPasswordModalClose, setPasswordModalOnClose] =
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx]: Already update the names! Please check.
https://github.com/reearth/reearth-visualizer/pull/1155/commits/65e8f09467a9f1f3023eebc74f254ec1b7938c5c

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+              </PasswordInputWrapper>
+            </PasswordWrapper>
+          </SettingsFields>
+        </Collapse>
+      </SettingsWrapper>
+
+      <PasswordModal
+        isVisible={!onPasswordModalClose}
+        passwordPolicy={passwordPolicy}
+        onClose={() => setPasswordModalOnClose(!onPasswordModalClose)}
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx]: Already changing to give the value directly!
https://github.com/reearth/reearth-visualizer/pull/1155/commits/280b8f1fb23710a13d0c1de3c3969bb9889b4f83

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+              value={informationData.email ? t(informationData.email) : ""}
+              appearance="readonly"
+              disabled
+            />
+
+            <PasswordWrapper>
+              <Typography size="body">{t("Password")}</Typography>
+              <PasswordInputWrapper>
+                <TextInput
+                  value={t("**********")}
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx]: Already delete useless translate!
https://github.com/reearth/reearth-visualizer/pull/1155/commits/90e2bc8a8a39fbd581458a520610f47b29614b4a

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+  lowSecurity?: RegExp;
+  medSecurity?: RegExp;
+  highSecurity?: RegExp;
+};
+
+type Props = {
+  isVisible: boolean;
+  onClose?: () => void;
+  passwordPolicy?: PasswordPolicy;
+  updatePassword?: ({
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx]: Already change updatePassword to onPasswordUpdate.
https://github.com/reearth/reearth-visualizer/pull/1155/commits/a42d912b14f614d1b765b9edc871c7c424b055ef

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx [hunk 18]
```diff
@@ -0,0 +1,200 @@
...
+          break;
+        case passwordPolicy?.lowSecurity?.test(password ?? ""):
+          setRegexMessage(t("That password is okay."));
+          break;
+        default:
+          setRegexMessage(t("That password confuses me, but might be okay."));
+          break;
+      }
+    },
+    [t, password, passwordPolicy] // eslint-disable-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx]: yes!  I deleted it.
https://github.com/reearth/reearth-visualizer/pull/1155/commits/103485c7cd5bf7d89311aad838af5c1edb00e574

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx (hunk 10 above)
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx]: Reversed to use the color from theme!
https://github.com/reearth/reearth-visualizer/pull/1155/commits/495dced0d6f017c0d41f2aabea02b61356c344af

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/common.tsx (hunk 12 above)
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/common.tsx]: deleted!
https://github.com/reearth/reearth-visualizer/pull/1155/commits/bfe46f3e8a94c6c98880f376c16256a5a55c26e1

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/common.tsx (hunk 13 above)
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/common.tsx]: deleted!
https://github.com/reearth/reearth-visualizer/pull/1155/commits/bfe46f3e8a94c6c98880f376c16256a5a55c26e1

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx (hunk 9 above)
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx]: Already added color for regexMessage. Please check.
https://github.com/reearth/reearth-visualizer/pull/1155/commits/a431d2297da2511f6772fac437ec327fd69dc38a

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx (hunk 11 above)
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx]: Already change font-weight to `regular`. And delete useless div, style component to use existed gap.
https://github.com/reearth/reearth-visualizer/pull/1155/commits/7b82f8cad5f29e78af4919388373919819c1ab73

File: web/src/beta/features/AccountAndWorkSpaceSetting/index.tsx [hunk 19]
```diff
@@ -0,0 +1,132 @@
...
+        path: `/settings/${tab.id}`
+      })),
+    [t]
+  );
+  const { meData, passwordPolicy, handleUpdateUserPassword } = useHook();
+  const { name, email } = meData;
+
+  return (
+    <Wrapper>
+      <Navbar workspaceId={meData.myTeam?.id} page="settings" />
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/index.tsx:47]: Yes, I will do it with other branch later

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx [hunk 20]
```diff
@@ -0,0 +1,98 @@
...
+  }: UpdatePasswordType) => Promise<void>;
+};
+
+const AccountSetting: FC<Props> = ({
+  passwordPolicy,
+  onUpdateUserPassword,
+  informationData
+}) => {
+  const t = useT();
+  const [passwordModalClosed, setPasswordModalClosed] = useState<boolean>(true);
```
Comment (mkumbobeaty) [inline web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx:33]:
```suggestion
  const [changePasswordModal, setChangePasswordModal] = useState<boolean>(false);
```
can you rename to match its functionality

File: web/src/beta/features/AccountAndWorkSpaceSetting/index.tsx [hunk 21]
```diff
@@ -0,0 +1,132 @@
...
+  backgroundColor: theme.bg[0],
+  ["*"]: {
+    boxSizing: "border-box"
+  },
+  ["* ::-webkit-scrollbar"]: {
+    width: "8px"
+  },
+  ["* ::-webkit-scrollbar-track"]: {
+    background: theme.relative.darker,
+    borderRadius: "10px"
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/index.tsx]: I has changed borderRadius to use theme, Please check!
https://github.com/reearth/reearth-visualizer/pull/1155/commits/eb65c3925b28e0f6232860a70d6a5773515a1d3e

File: web/src/beta/features/AccountAndWorkSpaceSetting/index.tsx [hunk 22]
```diff
@@ -0,0 +1,132 @@
...
+  },
+  ["* ::-webkit-scrollbar-thumb"]: {
+    background: theme.relative.light,
+    borderRadius: "4px"
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/index.tsx]: I has changed borderRadius to use theme, Please check!
https://github.com/reearth/reearth-visualizer/pull/1155/commits/eb65c3925b28e0f6232860a70d6a5773515a1d3e

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx [hunk 23]
```diff
@@ -0,0 +1,214 @@
...
+                size="body"
+                weight="bold"
+                color={regexMessageColor}
+              >
+                {regexMessage}
+              </PasswordMessage>
+            ) : undefined}
+          </PasswordField>
+          <PasswordField direction="column">
+            <Typography size="body" weight="bold">
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx]: I has deleted all no need bold and font style, Please check!
https://github.com/reearth/reearth-visualizer/pull/1155/commits/33bd09ddedc54c8a20e5fa3e1f9da6bdb9494f28

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx [hunk 24]
```diff
@@ -0,0 +1,214 @@
...
+              onChange={setPasswordConfirmation}
+              type="password"
+            />
+            {isMatchPassword ? (
+              <PasswordMessage
+                size="body"
+                weight="regular"
+                color={theme.dangerous.main}
+              >
+                <span>
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/PasswordModal/index.tsx]: I have deleted the unneeded span. Please check!
https://github.com/reearth/reearth-visualizer/pull/1155/commits/b48dd28e8a27b330e284d86623c7aee55a12e254

File: web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx (hunk 20 above)
Comment (ZTongci) [reply web/src/beta/features/AccountAndWorkSpaceSetting/innerPages/AccountSetting/index.tsx]: Change name to changePasswordModal. It's more simple to use. Please check.
https://github.com/reearth/reearth-visualizer/pull/1155/commits/528ac026693a7a96c8a8a43b11b70df4cdfe1c8c

**Review by airslice — APPROVED** [review]

**Review by mkumbobeaty — APPROVED** [review]

**Pull request merged**